django/utils/deprecation.py

```diff
@@ -90,6 +90,7 @@ def __init__(self, get_response):
         if get_response is None:
             raise ValueError('get_response must be provided.')
         self.get_response = get_response
+        self._get_response_is_coroutine = asyncio.iscoroutinefunction(self.get_response)
```

Comment:
It feels a bit funny doing the exact same check twice in two lines, here and then in `_async_check()` — if we can't reuse `_is_coroutine`, could setting this move into the `_async_check()` method?

Reply:
Good point, I moved it down to always setting `self._is_coroutine`